README.md

```diff
@@ -1,289 +1,608 @@
 
+### Resource Type Schema
+
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  name:         # string
+spec:
+  name:         # string, optional, default: .metadata.name
+
+  application:  # PodSpec-able resource ObjectReference
+    apiVersion: # string
+    kind:       # string
+    name:       # string
+    ...
+
+  service:      # Provisioned Service-able resource ObjectReference
+    apiVersion: # string
+    kind:       # string
+    name:       # string
+    ...
+
+  env:          # []EnvVar, optional
+  - name:       # string
+    value:      # string
+  ...
 ```
-$SERVICE_BINDINGS_ROOT/<service-id>/metadata/<persisted_metadata>
+
+### Example Resource
+
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: ServiceBinding
+metadata:
+  name: online-banking-to-account-service
+spec:
+  name: account-service
+
+  application:
+    apiVersion: apps/v1
+    kind:       Deployment
+    name:       online-banking
+
+  service:
+    apiVersion: com.example/v1alpha1
+    kind:       AccountService
+    name:       prod-account-service
+
+  env:
+  - name:  ACCOUNT_SERVICE_HOST
+    value: ((host))
+  - name:  ACCOUNT_SERVICE_USERNAME
+    value: ((username))
+  - name:  ACCOUNT_SERVICE_PASSWORD
+    value: ((password))
+  - name:  ACCOUNT_SERVICE_URI
+    value: ((accountServiceUri))
 ```
 
-The recommended set of metadata will vary dependending on implementations and platforms, but two **RECOMMENDED** keys are:
-* kind
-* provider
+## Synthetic Provisioned Service
+
+There are some situations where there is an arity mismatch between a collection of Kubernetes resources representing a system and an application that wishes to consume them.  The solution to this problem is to create a synthetic Provisioned Service that is a composite of multiple other resources.
+
+A Synthetic Provisioned Service resource **MUST** define a `.spec.services` which is an array of `ObjectReference`s to a resource.
+
+A Synthetic Provisioned Service resource **MUST** define a `.spec.mappings` which is an array of `Mapping` objects. A `Mapping` object **MUST** define `name` and `value` entries. The value of a `Mapping` **MAY** contain zero or more tokens beginning with `((`, ending with `))`, and encapsulating a [JSON Path](https://kubernetes.io/docs/reference/kubectl/jsonpath/) to an entry on a resource defined in `services`. The value of this `Secret` entry **MUST** be substituted into the original value string, replacing the token. Once all tokens have been substituted, the new value **MUST** be added to the binding `Secret` exposed by the resource.
```

Comment:
To help with the adoption of the spec, what do you think about letting users choose between Go templates and JSONPath?
Go template is familiar to people with Helm background and also available in `kubectl`.

Reply:
There was a Go Template section originally, but I couldn't work it in because it wasn't particularly well defined.  As in, there wasn't enough information to even make a coherent example.  I'm happy to include it, but I'd need a better example/spec to include.  I'd recommend a subsequent PR, but if you can get something into a comment here, I would include it directly 😁

Reply:
Sounds good, I can do a separate PR for adding the Go Template here.

Reply:
There were a couple of scenarios where I had found composition using Go templates / jsonpath particularly helpful.

In the awkward situation where the the provisioned service doesn't actually have a provisioned service secret, the custom environment variable / mapping we were creating the mapping through Go templates or jsonpath 

```
 - name: LANGUAGE_TRANSLATOR_URL
    value: '{{ index .status.secretName "url" }}'
```

This is basically the escape hatch if the provider has neither provided a duck type nor any annotations/descriptors. So this was useful for both single and multiple services.